packages/flutter_driver/lib/src/common/text_input_action.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'enum_util.dart';
+import 'message.dart';
+
+EnumIndex<TextInputAction> _textInputActionIndex =
+    EnumIndex<TextInputAction>(TextInputAction.values);
+
+/// A Flutter Driver command that send a text input action.
+class SendTextInputAction extends Command {
+  /// Creates a command that enters text into the currently focused widget.
+  const SendTextInputAction(this.textInputAction, {super.timeout});
+
+  /// Deserializes this command from the value generated by [serialize].
+  SendTextInputAction.deserialize(super.json)
+      : textInputAction =
+            _textInputActionIndex.lookupBySimpleName(json['action']!),
+        super.deserialize();
+
+  /// The [TextInputAction]
+  final TextInputAction textInputAction;
+
+  @override
+  String get kind => 'send_text_input_action';
+
+  @override
+  Map<String, String> serialize() => super.serialize()
+    ..addAll(<String, String>{
+      'action': _textInputActionIndex.toSimpleName(textInputAction),
+    });
+}
+
+/// An action the user has requested the text input control to perform.
+///
+/// This class is identical to [TextInputAction](https://api.flutter.dev/flutter/services/TextInputAction.html).
+/// This class is cloned from `TextInputAction` and must be kept in sync. The cloning is needed
+/// because importing is not allowed directly.
+enum TextInputAction {
```

Comment:
Do we do this for other enums in this package? if so, I'd be worried about keeping them in sync. It would be great to have a test that checked that they had the same entries at least

Reply:
hi @jonahwilliams 
i tried to use the same entries, but build error because that package include UI, so it can't compile in server side. 

Do you have any suggestion for me ? Thanks.

Reply:
I don't mean using the same enum, I mean writing a test that verifies that these enums have the same values. i.e. in the test open the files and check the values

Reply:
@jonahwilliams , i added unit-test for do it, pls help me review again.